components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/api/PasswordRecoveryApiV1.java

```diff
@@ -0,0 +1,194 @@
+ */
+public class PasswordRecoveryApiV1 {
+
+    String basePath = IdentityManagementEndpointUtil.buildEndpointUrl(IdentityManagementEndpointConstants
+            .UserInfoRecovery.RECOVERY_API_V1_RELATIVE_PATH);
+    private ApiClient apiClient;
+
+    public PasswordRecoveryApiV1() {
+
+        this(Configuration.getDefaultApiClient());
+    }
+
+    public PasswordRecoveryApiV1(ApiClient apiClient) {
+
+        this.apiClient = apiClient;
+    }
+
+    public ApiClient getApiClient() {
+
+        return apiClient;
+    }
+
+    public void setApiClient(ApiClient apiClient) {
+
+        this.apiClient = apiClient;
+    }
+
+    /**
+     * This API can be used to initiate recovering forgotten password.
+     *
+     * @param recoveryInitRequest Password recovery initiating request. (required)
+     * @param tenantDomain        Tenant Domain which user belongs. Default &#x60;carbon.super&#x60; (optional)
+     * @param headers             If reCaptcha respond is found, embedded in request header. (optional)
+     * @return Account recovery options response object.
+     * @throws ApiException if fails to make API call.
+     */
+    public List<AccountRecoveryType> initiatePasswordRecovery(RecoveryInitRequest recoveryInitRequest, String tenantDomain,
+                                                              Map<String, String> headers) throws ApiException {
+
+        String localVarPath = "/password/init".replaceAll("\\{format\\}", "json");
+        return initiateRecovery(recoveryInitRequest, tenantDomain, null, localVarPath);
+    }
+
+    /**
+     * This API is used to recover password via selected recovery option.
+     *
+     * @param recoveryRequest   Recovery request. (required)
+     * @param tenantDomain      Tenant Domain which user belongs. Default &#x60;carbon.super&#x60; (optional)
+     * @param headers           Any additional headers to be embedded. (optional)
+     * @return Recovery response.
+     * @throws ApiException if fails to make API call.
+     */
+    public RecoveryResponse recoverPassword(RecoveryRequest recoveryRequest, String tenantDomain,
+                                            Map<String, String> headers) throws ApiException {
+
+        String localVarPath = "/password/recover".replaceAll("\\{format\\}", "json");
+        return recover(recoveryRequest, tenantDomain, headers, localVarPath);
+    }
+
+    /**
+     * This API can be used to initiate recovering forgotten password/username.
+     *
+     * @param recoveryInitRequest Recovery initiating request. (required)
+     * @param tenantDomain        Tenant Domain which user belongs. Default &#x60;carbon.super&#x60; (optional)
+     * @param headers             If reCaptcha respond is found, embedded in request header. (optional)
+     * @param localVarPath        Endpoint path.
+     * @return Account recovery options response object.
+     * @throws ApiException if fails to make API call.
```

Comment:
Shall we start the comment with an uppercase letter. WDYT?

Reply:
Addressed in https://github.com/wso2/carbon-identity-framework/pull/3171/commits/09b57785a1049994143d57f2e0620c3f88376e14